### PR TITLE
Mining tweaks, fixes and balance changes

### LINKED ISSUE #18796: Somehow my corpse got sent into nullspace because it didnt drop from the legion made with my  corpse
Thats... all i know. I couldn't ghost from my body and they told me it was in the nullspace void.

### PR DESCRIPTION
**Pull request by Core0verload**
**Tweaks and QoL changes:**
- Kinetic accelerator now supports seclite attachment.
- Miners now have a special heavy duty headset, it differs from regular supply headset only in sprite.
- Miner's roundstart crowbar now spawns in box instead of backpack.
- Preserved legion core no longer sparks and has a telling desc.

miner

**Balance changes:**
- All cyborgs are now ash-proof. Mining cyborg starts with ash-proof sprite to fit dark mining style. All borgs are superior to humans at their module's job, it makes sense for mining cyborg to come with ash-proof hull by default - after all, mining borgs can't deploy shelters to sit out ash storms.
- Ash-proof plating upgrade was replaced with Lavaproof tracks. This upgrade allows mining cyborgs to roam lava safely.
- Mining cyborg module now includes a tiny welding tool for basic self-repair. This welding tool holds 4 times less fuel than the one used by engineering cyborgs. It also includes a small fire extinguisher.
- Miners start with a box of 3 brute and 1 burn paths instead of a single brute patch. This compensates for recent path nerf that dropped amount of reagents in single path more than in two.
- Explorer's belt slot amount is increased by one, while mining belt crafted from sinew still has 5 slots.

**Fixes:**
- Updates miner's corpse and mining wardrobe with new lavaland gear.
- Fixes envy's knife path.
- Fixes wing potion's misguided racism.
- Fixes #18796. Fixes legion's nullspace corpse drop bug and two related runtimes.

**Token:** I think included fixes count as one.

:cl: CoreOverload
rscadd: Kinetic accelerator now supports seclite attachment.
rscadd: All cyborgs have been outfitted with ash-proof plating.
rscadd: Mining cyborg module now includes a welder and a fire extinguisher.
rscadd: Mining cyborgs now have a new upgrade: lavaproof tracks.
/:cl:

### CODE REVIEW

**Comment by Durkel** [comment]
hugbox mining.

**Comment by Core0verload** [comment]
Not hugbox at all. Most of the changes affect mining cyborgs, who tend to die horribly even more often than regular miners because they lack armor. They are supposed to be superior to human miners.

Paths change is because of recent paths nerf. 3 brute paths now are like 1 brute path before.

**Comment by yackemflam** [comment]
I really don`t like where mining is going atm

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 1]
```diff
@@ -213,7 +213,7 @@
 		if("Miner")
 			module = new /obj/item/weapon/robot_module/miner(src)
 			hands.icon_state = "miner"
-			icon_state = "minerborg"
+			icon_state = "ashborg"
```
Comment (KorPhaeron) [inline code/modules/mob/living/silicon/robot/robot.dm]: Why not let them choose like service borgs do?

**Comment by PKPenguin321** [comment]
pimp my miner

File: code/game/objects/items/robot/robot_upgrades.dm [hunk 2]
```diff
@@ -233,7 +233,7 @@
 
 	return 1
 
-/obj/item/borg/upgrade/ashplating
+/*/obj/item/borg/upgrade/ashplating
```
Comment (KorPhaeron) [inline code/game/objects/items/robot/robot_upgrades.dm]: Remove unused code, don't comment it out

File: code/game/objects/items/weapons/storage/belt.dm [hunk 3]
```diff
@@ -173,25 +173,28 @@
...
-		/obj/item/weapon/ore/bluespace_crystal,
-		/obj/item/weapon/reagent_containers/food/drinks,
-
-
+		/obj/item/weapon/storage/pill_bottle,
+		/obj/item/weapon/ore,
+		/obj/item/weapon/reagent_containers/food/drinks
 		)
 
+/obj/item/weapon/storage/belt/mining/vendor
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/belt.dm]: This is already the most chosen item from the vendor, I don't think it needs a buff

File: code/modules/mob/living/silicon/robot/robot.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/mob/living/silicon/robot/robot.dm]: The same reason we don't let miners chose between lavaland-styled gear and old gear.

File: code/game/objects/items/weapons/storage/belt.dm (hunk 3 above)
Comment (Core0verload) [inline code/game/objects/items/weapons/storage/belt.dm]: fuck, this thing is so large it fails to load in my browser

Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/belt.dm]: Survival_Capsule_and_Explorer's_Webbing	1451	49.23%
Advanced_Scanner	1071	36.34%
Resonator	166	5.63%
Mining_Drone	151	5.12%
Survival_Capsule_and_Explorer_Belt	108	3.66%

Comment (Core0verload) [inline code/game/objects/items/weapons/storage/belt.dm]: Seems like the data is old. Advanced Scanner and Resonator are now a single item. I think I'll add extra paths to them.

Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/belt.dm]: Page auto-generated by cheri on 16 June 2016 at 18:13:57 in CEST (GMT +0200)

It's from like 2 days ago

**Comment by GunHog** [comment]
@KorPhaeron Could you give me something else I could do with the plating upgrade if this goes through? It seems a shame to let WJ's sprite (for the item) go to waste. Maybe a high-end lava upgrade?

**Comment by Xhuis** [comment]
Legions don't drop legion skulls right now, meaning the stuffed legion is impossible to make. It might not be a bad idea to make them droppable and maybe breakable into bone if the miner has no interest in cooking with it.

**Comment by Core0verload** [comment]
I don't like both legion skull and stuffed legion. IMO it was a bad idea.

**Comment by yackemflam** [comment]
Why give miners special snowflake headsets?

**Comment by GunHog** [comment]
It is just a sprite, relax.

**Comment by Core0verload** [comment]
Just drop the code somewhere and I'll add it.

**Comment by GunHog** [comment]
Just change the weather immunity from "ash" to "lava" and raise the tech/mineral reqs. I do not know an easy way to work from your branch.

**Comment by Core0verload** [comment]
Done.

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 4]
```diff
@@ -430,7 +430,7 @@
 			if(C.wear_mask)
 				C << "<span class='notice'>It's pretty hard to drink something with a mask on!</span>"
 			else
-				if(ishumanbasic(C)) //implying xenoshumans are holy
+				if(!ishumanbasic(C)) //implying xenoshumans are holy
 					C << "<span class='notice'>You down the elixir, noting nothing else but a terrible aftertaste.</span>"
 				else
 					C << "<span class='userdanger'>You down the elixir, a terrible pain travels down your back as wings burst out!</span>"
```
Comment (KazeEspada) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Doesn't this mean that only special snowflake humans can get wings?

Comment (Core0verload) [inline code/modules/mining/lavaland/necropolis_chests.dm]: No. This means that only humans can get wings. Lizards, golems, podmen and whatever nonhuman bullshit we currently have get nothing.

Before, only nonhuman races were getting wings.

File: code/modules/research/designs/mechfabricator_designs.dm [hunk 5]
```diff
@@ -658,11 +658,11 @@
...
-	name = "Cyborg Upgrade (Ash Storm Plating)"
-	id = "borg_upgrade_ashplating"
+/datum/design/borg_upgrade_lavaproof
+	name = "Cyborg Upgrade (Lavaproof Tracks)"
+	id = "borg_upgrade_lavaproof"
 	build_type = MECHFAB
-	build_path = /obj/item/borg/upgrade/ashplating
+	build_path = /obj/item/borg/upgrade/lavaproof
 	req_tech = list("plasmatech" = 4, "materials" = 4, "engineering" = 4)
 	materials = list(MAT_METAL = 8000, MAT_PLASMA = 10000)
```
Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: I think buying plasma inmunity is dumb in the first place but I think making it out of the most flammable material in the game is extra dumb

**Comment by Hatfish** [comment]
the QoL changes are more or less fine, but the "balance" changes will make lavaland piss-easy.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 5 above)
Comment (RemieRichards) [inline code/modules/research/designs/mechfabricator_designs.dm]: you mean lava immunity, not plasma immunity.

Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: no bully

File: code/game/objects/items/weapons/storage/belt.dm [hunk 6]
```diff
@@ -146,7 +146,7 @@
 	desc = "A versatile chest rig, cherished by miners and hunters alike."
 	icon_state = "explorer1"
 	item_state = "explorer1"
-	storage_slots = 5
+	storage_slots = 6
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/belt.dm]: ?

**Comment by anconfuzedrock** [comment]
all i care is that headset looks kinda neat and I have wanted a flashlight on my ka for ages

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 4 above)
Comment (MMMiracles) [inline code/modules/mining/lavaland/necropolis_chests.dm]: The potion was made in mind with only humans actually getting effects from it, because the implication of xenohumans being holy enough is actual heresy.

I don't know who last touched it but it worked as intended beforehand.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 5 above)
Comment (GunHog) [inline code/modules/research/designs/mechfabricator_designs.dm]: Plasteel, made of plasma and iron, completely fireproof. I recommended to increase the tech/mineral reqs due to lava immunity being a lot more powerful than ash storm immunity! I will even beg WJ to make an even sexier lava sprite!

Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: Stun arm + dragging into lava might be kinda unfun as well

Comment (KorPhaeron) [inline code/modules/research/designs/mechfabricator_designs.dm]: Maybe we should end this token thing, it discourages people from atomizing prs

File: code/game/objects/items/weapons/storage/belt.dm (hunk 6 above)
Comment (Core0verload) [inline code/game/objects/items/weapons/storage/belt.dm]: See changelog.

**Comment by SidonCC** [comment]
All cyborgs are ash proof?

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 5 above)
Comment (Core0verload) [inline code/modules/research/designs/mechfabricator_designs.dm]: Atomizing PRs is annoying even without tokens. Instead of having to get 1 large PR merged, you had to get 20 small PRs merged.

**Comment by Jordie0608** [comment]
Your changelog doesn't mention a lot of those changes

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm [hunk 7]
```diff
@@ -55,10 +55,10 @@
 /mob/living/simple_animal/hostile/megafauna/ex_act(severity, target)
 	switch (severity)
 		if (1)
-			adjustBruteLoss(250)
+			adjustBruteLoss(400)
```
Comment (KorPhaeron) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: I still dont understand the point of having a larger health pool if you're just going to scale the damage it takes to match it

Comment (Core0verload) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: I already said it more than once, and I say it again. Using bombs and risking getting caught in explosion for a chance to deal 50 damage is simply not worth it when you can just shot the boss with KA twice for 80 damage with zero risk.

**Comment by Core0verload** [comment]
@Jordie0608, it's all in PR post. And AFAIK there is a policy about not marking lavaland changes in changelog because "unexplored".

**Comment by Jordie0608** [comment]
There is no such policy, only lazy coders, and even if there was none of the unmentioned content in your PR is potentially explorable that would warrant it being kept a secret.

File: code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm (hunk 7 above)
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm]: For reference, the previous pr on this: #18531

**Comment by optimumtact** [comment]
Why did you add the explosion changes? Wasn't that pr denied and closed before? Seriously dude, you got github banned once for playing games with Cheridan, why do you keep toeing the line?

**Comment by Core0verload** [comment]
First, I was never banned on tg, second, PR was denied because of megafauna immunity removal, not because of explosion changes. Fuck off.

**Comment by Core0verload** [comment]
@ChangelingRain, remove "Map PR with no picture" tag. The only map change is path update, doing pictures for path changes is simply stupid.

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 4 above)
Comment (KazeEspada) [inline code/modules/mining/lavaland/necropolis_chests.dm]: Thats dumb... i wanna become a dragon lizardman.

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 5 above)
Comment (optimumtact) [inline code/modules/research/designs/mechfabricator_designs.dm]: The reason your pr is taking so long to merge is all the crap you have in it.

I would have happily merged a pr containing the fixes,  a pr containing QOL changes and a pr containing the borg stuff ages ago, as it is, I'm unwilling to touch it because of the explosion changes because I know at least some people will get mad about it.

File: code/modules/mining/lavaland/necropolis_chests.dm (hunk 4 above)
Comment (GunHog) [inline code/modules/mining/lavaland/necropolis_chests.dm]: I could totally steal @WJohn's old dragon wing sprites to apply to lizards when they drink it. So, Lizard drinkers = Dragons, Human drinkers = Angels!

File: code/modules/research/designs/mechfabricator_designs.dm (hunk 5 above)
Comment (GunHog) [inline code/modules/research/designs/mechfabricator_designs.dm]: Yes, please remove the explosion changes, we do not need bomb-cheesing tactics holding up the PR. Make another PR for that!

**Comment by Kromgar** [comment]
Hey changeling. Make mining hardsuits ashstorm proof and have fire immunity like the explorers webbing

**Comment by AlexCrimson** [comment]
Are non-KA Mining tools ever going to get a buff? 9/10 people just use the KA now. Its rather boring.

**Comment by Core0verload** [comment]
I buffed plasma cutter recently. Maybe I'll add a basic cutter to mining vendor too.

**Comment by KorPhaeron** [comment]
atomize your prs please

**Comment by Core0verload** [comment]
Conflicts resolved, and explosion tweaks are moved to another PR.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 8]
```diff
@@ -68,6 +68,15 @@
 	..()
 	new /obj/item/device/radio/off(src)
 
+/obj/item/weapon/storage/box/survival_mining/New()
+	..()
+	new /obj/item/clothing/mask/breath(src)
+	new /obj/item/weapon/tank/internals/emergency_oxygen/engi(src)
+	new /obj/item/weapon/crowbar/red(src)
+	new /obj/item/weapon/reagent_containers/pill/patch/styptic(src)
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/storage/boxes.dm]: Did you mean to remove this now that they get a bottle?

File: code/modules/mining/machine_vending.dm [hunk 9]
```diff
@@ -158,16 +158,18 @@
...
 		return
 	switch(selection)
 		if("Survival Capsule and Explorer's Webbing")
-			new /obj/item/weapon/storage/belt/mining(src.loc)
-			new /obj/item/weapon/survivalcapsule(src.loc)
+			new /obj/item/weapon/storage/belt/mining/vendor(src.loc)
 		if("Resonator and Advanced Scanner")
 			new /obj/item/weapon/resonator(src.loc)
 			new /obj/item/device/t_scanner/adv_mining_scanner(src.loc)
+			new /obj/item/weapon/storage/pill_bottle/mining(src.loc)
```
Comment (KorPhaeron) [inline code/modules/mining/machine_vending.dm]: Starting with 7 patches seems excessive

Comment (Core0verload) [inline code/modules/mining/machine_vending.dm]: It's voucher only after all.

File: code/game/objects/items/weapons/storage/boxes.dm (hunk 8 above)
Comment (Core0verload) [inline code/game/objects/items/weapons/storage/boxes.dm]: yeah, will remove

**Comment by KorPhaeron** [comment]
conflict

**Comment by Core0verload** [comment]
no conflict

**Comment by optimumtact** [comment]
Woop woop better back it up
